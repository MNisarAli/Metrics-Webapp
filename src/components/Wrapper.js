import PropTypes from 'prop-types';
import { HiOutlineArrowCircleRight } from 'react-icons/hi';

const Wrapper = ({ wrapper }) => {
  // Destructure the properties from the 'wrapper' object.
  const {
    name = '',
    population = '',
    map = '',
  } = wrapper;
  return (
    <>
      <div className="right-arrow">
        <HiOutlineArrowCircleRight />
      </div>
      <div className="content-wrapper">
        <img className="map" src={map} alt={name} />
        <div className="content">
          <h3 className="sub-heading">{name}</h3>
          {String(population)}
        </div>
      </div>
    </>
  );
};

Wrapper.propTypes = {
  // Validate the shape of the 'wrapper' object.
  wrapper: PropTypes.shape({
    name: PropTypes.string,
    population: PropTypes.oneOfType([
      PropTypes.string,
      PropTypes.number,
    ]),
    map: PropTypes.string,
  }),
};

Wrapper.defaultProps = {
  wrapper: {
    name: '',
    population: '',
    map: '',
  },
};

export default Wrapper;
