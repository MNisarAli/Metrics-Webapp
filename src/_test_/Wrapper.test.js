import React from 'react';
import { render } from '@testing-library/react';
import { Provider } from 'react-redux';
import renderer from 'react-test-renderer';
import { BrowserRouter } from 'react-router-dom';
import store from '../Redux/configureStore';
import '@testing-library/jest-dom';
import Wrapper from '../components/Wrapper';

describe('Wrapper component', () => {
  test('renders correctly', () => {
    const tree = renderer.create(
      <Provider store={store}>
        <BrowserRouter>
          <Wrapper />
        </BrowserRouter>
      </Provider>,
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });

  test('renders the Wrapper component with non-default props', () => {
    const { getByAltText, getByText } = render(
      <Wrapper wrapper={{ name: 'Test Wrapper', population: '1,000,000', map: 'test-map.png' }} />,
    );
    const wrapperMap = getByAltText('Test Wrapper');
    const wrapperName = getByText('Test Wrapper');
    expect(wrapperMap).toBeInTheDocument();
    expect(wrapperName).toBeInTheDocument();
  });
});
