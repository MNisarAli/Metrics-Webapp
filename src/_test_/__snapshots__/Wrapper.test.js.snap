// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Wrapper component renders correctly 1`] = `
Array [
  <div
    className="right-arrow"
  >
    <svg
      fill="none"
      height="1em"
      stroke="currentColor"
      strokeWidth="0"
      style={
        Object {
          "color": undefined,
        }
      }
      viewBox="0 0 24 24"
      width="1em"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M13 9l3 3m0 0l-3 3m3-3H8m13 0a9 9 0 11-18 0 9 9 0 0118 0z"
        strokeLinecap="round"
        strokeLinejoin="round"
        strokeWidth="2"
      />
    </svg>
  </div>,
  <div
    className="content-wrapper"
  >
    <img
      alt=""
      className="map"
      src=""
    />
    <div
      className="content"
    >
      <h3
        className="sub-heading"
      />
    </div>
  </div>,
]
`;
